linux_docker_resources/entry_point.sh

```diff
@@ -85,6 +85,4 @@ sed -i -e "s/rosbuild:x:$ORIG_GID:/rosbuild:x:$GID:/" /etc/group
 chown -R ${UID}:${GID} "${ORIG_HOME}"
 echo "done."
 
-cd "@workdir"
-
```

Comment:
On reflection, removing this bit of the script actually makes bind-mounting it long-term a lot less important. I need to bind-mount it in #640 because I can't change the embedded script which is causing me problems with the workdir but we may not need to generally. I'm open to thoughts on pulling the bind-mount out of the permanent change entirely.

Reply:
What purpose does this line serve, anyway? Doesn't the `WORKDIR` entry in the Dockerfile already ensure this is the case?

If we can really drop this, then I'd rather see the `ADD` kept in favor of more bind mounts.

Reply:
> What purpose does this line serve, anyway? 

It's redundant with `WORKDIR` (or `--workdir` in the new world order) and I'm pretty sure it was unnecessary when introduced in #248. I think the only reason to keep anything would a construct like

```
#!/bin/sh
PWD_BACKUP=$(pwd)
# script contents
cd $PWD_BACKUP
```
to make sure that a wayward script change doesn't change the directory when it shouldn't. However, I don't feel the need. I had this to say about it in ebf6e079ab10afddbba824ae1f34378ac459ea52 

> By eliminating this we remove a source of job-specific content in the
scripts and Dockerfile contents for CI.
> 
> By removing this we're also committing ourselves to being extra vigilant
that behavior inside the entry_point.sh script does not change the
directory context without restoring it. If there was concern about that
then an alternative would be to add

Reply:
16af6de3426ea2f5f2bcc4262990e5195c838669 effectively restores the original Dockerfile with one difference, added to a line comment on site.